tpe/plugin/src/SimulationFeatures.cc

```diff
@@ -98,6 +98,40 @@ void SimulationFeatures::Write(ChangedWorldPoses &_changedPoses) const
     }
   }
 
+  // Iterate over models to make sure link velocities for moving models
+  // are calculated and sent
+  for (const auto &[id, info] : this->models)
+  {
+    // make sure the model exists
+    if (info)
+    {
+      if (info->model->GetStatic())
+        continue;
+      const auto nextPose = info->model->GetPose();
+      // Note, now prevLinkPoses also containes prevModelPoses
+      // Data structure has been kept to avoid breaking ABI
+      auto iter = this->prevLinkPoses.find(id);
+
+      // If the models's pose is new or has changed, calculate and add all
+      // the children links' poses to the output poses
+      if ((iter == this->prevLinkPoses.end()) ||
+          !iter->second.Pos().Equal(nextPose.Pos(), 1e-6) ||
+          !iter->second.Rot().Equal(nextPose.Rot(), 1e-6))
+      {
+        for (const auto &linkEnt : info->model->GetChildren())
+        {
+          WorldPose wp;
+          wp.pose = linkEnt.second->GetPose();
+          wp.body = linkEnt.second->GetId();
+          _changedPoses.entries.push_back(wp);
```

Comment:
do you know if we'll end up pushing duplicate link entries into `_changedPoses`? I see that `_changedPoses` has a fixed size set [here](https://github.com/ignitionrobotics/ign-physics/blob/542f56d4d4e34daa3c6bbd2f6da44be762034cef/tpe/plugin/src/SimulationFeatures.cc#L72):

```
_changedPoses.entries.reserve(this->links.size());
```

So could this push more entries into the vector that the reserved size?